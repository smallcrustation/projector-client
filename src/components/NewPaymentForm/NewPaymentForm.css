.NewPaymentForm{

}

.NewPaymentForm__form{
  margin-top: 35%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.NewPaymentForm__label{
  font-size: 30px;
}