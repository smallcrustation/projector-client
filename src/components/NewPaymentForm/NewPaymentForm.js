import React from 'react'
import PaymentsApiService from '../../services/payments-api-service'
import { ProjectContext } from '../../contexts/ProjectContext'
import NumberFormat from 'react-number-format'

import './NewPaymentForm.css'

export default class NewPaymentForm extends React.Component {
  static contextType = ProjectContext
  state = { error: null }

  handleSubmit = async e => {
    e.preventDefault()

    const payment_amount = e.target.payment.value.replace(/[$,]/g, '')
    e.target.payment.value = ''

    const newPayment = {
      payment_number: this.context.payments
        ? this.context.payments.length + 1
        : 1,
      total_amount: payment_amount,
      project_id: this.context.project.id
    }

    try {
      // eslint-disable-next-line no-unused-vars
      const madePayment = await PaymentsApiService.postPayment(newPayment)
      this.props.onSuccessfulPayment()
    } catch (err) {
      this.setState({ error: err.error })
    }
  }

  render() {
    const error = this.state.error
    // ADD if else to go back to projects if !this.context.project
    return (
      <div className="NewPaymentForm">
        {!this.context.project ? (
          ''
        ) : (
          <h3 className="low-header">{this.context.project.project_name}</h3>
        )}
        <h3>Payment Request</h3>
        <form className="NewPaymentForm__form" onSubmit={this.handleSubmit}>
          <div role="alert">{error &&<p className="error">{error}</p>}</div>
          <label htmlFor="payment" className="NewPaymentForm__label">
            Amount
          </label>
          <NumberFormat
            name="payment"
            id="payment"
            className="dollar-input"
            placeholder="$0.00"
            thousandSeparator={true}
            prefix={'$'}
            decimalScale={2}
            fixedDecimalScale={2}
            required
          />

          <input type="submit" value="Submit Payment" />
        </form>
      </div>
    )
  }
}
