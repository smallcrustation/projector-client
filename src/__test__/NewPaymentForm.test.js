import React from 'react'
import ReactDOM from 'react-dom'

import NewPaymentForm from '../components/NewPaymentForm/NewPaymentForm'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<NewPaymentForm onSuccessfulPayment='' />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<NewPaymentForm onSuccessfulPayment=''/>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
