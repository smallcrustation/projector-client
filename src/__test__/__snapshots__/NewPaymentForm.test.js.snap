// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `""`;
